enhancements/ingress/gateway-api-crd-life-cycle-management.md

```diff
@@ -0,0 +1,496 @@
+- Describe the issue of dead fields and potential risks thereof.
+
+### Non-Goals
+
+- Automatically replace incompatible CRDs that some other agent installed.
+- Provide a pre-upgrade check in OpenShift 4.18 for incompatible CRDs.
+- Provide an explicit override for the cluster-admin to take CRD ownership.
+- Implement an admission webhook to block updates to dead fields.
+- Solve CRD life-cycle management in OLM.
+- Solve OLM subscription management.
+- Solve CRD life-cycle management for OSSM or Istio resources.
+
+## Proposal
+
+### Life-cycle management
+
+OpenShift's Ingress Operator monitors for the presence of the Gateway API CRDs
+and uses [Server-Side Apply](https://kubernetes.io/docs/reference/using-api/server-side-apply/) to track ownership using the following logic:
+
+- If CRDs are absent, the Ingress Operator installs the appropriate version.
+- If CRDs are present with the Ingress Operator as owner:
+  - If they are at the appropriate version, the operator does nothing.
+  - Else, the operator updates the CRDs to the appropriate version.
+- If CRDs are present with some other owner:
+  - If CRDs are at an unexpected version, the operator signals a degraded state.
+  - Else, _TODO: Do we signal degraded, or what?_
+
+Note that the appropriate version for the CRDs depends on the version of
+OpenShift Service Mesh (OSSM) that the Ingress Operator has pinned in a given
+OpenShift release.  Specifically, the CRDs SHOULD be the version corresponding
+to the version of Istio in that OSSM version; they MUST be of some version that
+is compatible and that we have tested with that version of Istio.  That is, in
+order for some version of the CRDs to be allowed to be accepted, Red Hat MUST
+run the upstream conformance tests and downstream end-to-end tests with the
+specific combination of OSSM version that the operator installs and CRD version
+that is desired to be installed.
+
+These CRDs are necessary for the Gateway API feature to work at all.  Ideally,
+the Ingress Operator is the only agent that attempts to manage the life-cycle of
+these CRDs (that is, creating, updating, or deleting them).  However, in some
+contingencies, this is not the case:
+
+- A cluster-admin can install the CRDs before upgrading to OpenShift 4.19.
+- A layered product or third-party controller could be actively managing them.
+
+In any scenario in which some agent other than the Ingress Operator has
+installed the CRDs, the Ingress Operator cannot trivially infer the intent and
+potential active use of the existing CRDs.  Overwriting the existing versions
+represents a risk of breaking workload or violating the end-user's expectations.
+Thus the Ingress Operator will only detect and warn about these scenarios.
+
+The Ingress Operator's warning will take the form of, at a minimum, setting the
+`Degraded` status condition's status to `True` with a descriptive message on the
+ingress clusteroperator.  
```

Comment:
At some point we will need to explain how an admin gets out of this, TODO delete this comment if I find that later

Reply:
I figure it can be documented under "Workflow Description" or "Support Procedures".  I can add a cross-reference here.

Reply:
That would be helpful, thank you